cub/test/catch2_test_device_for_utils.cu

```diff
@@ -99,8 +99,8 @@ void test()
 
 C2H_TEST("Device for utils correctly detect value overloads", "[for][device]")
 {
-  test<int>();
-  test<double>();
+  ::test<int>();
```

Comment:
Why do we need to fully qualify here?

Reply:
Because `C2H_TEST` was changed to a Catch2 Fixture, each unit test function becomes a member function of some test class and  there is a new base class that every test class inherits from. IIRC, calling `test` would be ambiguous with a Catch2 function here.